panoptes_client/subject.py

```diff
@@ -252,18 +268,23 @@ def add_location(self, location):
 
         try:
             media_data = f.read()
-            if MEDIA_TYPE_DETECTION == 'magic':
+            if manual_mimetype is not None:
+                media_type = manual_mimetype
+            elif MEDIA_TYPE_DETECTION == 'magic':
                 media_type = magic.from_buffer(media_data, mime=True)
             else:
-                media_type = imghdr.what(None, media_data)
+                media_type = mimetypes.guess_type(location)[0]
                 if not media_type:
                     raise UnknownMediaException(
                         'Could not detect file type. Please try installing '
                         'libmagic: https://panoptes-python-client.readthedocs.'
                         'io/en/latest/user_guide.html#uploading-non-image-'
                         'media-types'
                     )
-                media_type = 'image/{}'.format(media_type)
+
+            if media_type not in ALLOWED_EXTENSIONS.values():
```

Comment:
If this is the only application of `ALLOWED_EXTENSIONS`, should we change this to `ALLOWED_MIME_TYPES` and just use a list of the MIME values rather than `ALLOWED_EXTENSIONS.values()`?